gcloud-java-bigquery/src/test/java/com/google/gcloud/bigquery/DatasetTest.java

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.bigquery;
+
+import static org.easymock.EasyMock.createStrictMock;
+import static org.easymock.EasyMock.expect;
+import static org.easymock.EasyMock.replay;
+import static org.easymock.EasyMock.verify;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.assertTrue;
+
+import com.google.common.collect.ImmutableList;
+import com.google.gcloud.Page;
+import com.google.gcloud.PageImpl;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Iterator;
+
+public class DatasetTest {
+
+  private static final DatasetId DATASET_ID = DatasetId.of("dataset");
+  private static final DatasetInfo DATASET_INFO = DatasetInfo.builder(DATASET_ID).build();
+  private static final Field FIELD = Field.of("FieldName", Field.Type.integer());
+  private static final Iterable<BaseTableInfo> TABLE_INFO_RESULTS = ImmutableList.of(
+      TableInfo.builder(TableId.of("dataset", "table1"), Schema.of(FIELD)).build(),
+      ViewInfo.builder(TableId.of("dataset", "table2"), "QUERY").build(),
+      ExternalTableInfo.builder(TableId.of("dataset", "table2"),
+          ExternalDataConfiguration.of(ImmutableList.of("URI"), Schema.of(), FormatOptions.csv()))
+          .build());
+
+  private BigQuery bigquery;
+  private Dataset dataset;
+
+  @Before
+  public void setUp() throws Exception {
+    bigquery = createStrictMock(BigQuery.class);
+    dataset = new Dataset(bigquery, DATASET_INFO);
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    verify(bigquery);
+  }
+
+  @Test
+  public void testInfo() throws Exception {
```

Comment:
Nit: if we have a test for this, should we also have a test for the function `bigquery()`?